core/src/main/java/org/apache/spark/shuffle/sort/io/LocalDiskShuffleDataIO.java

```diff
@@ -35,12 +37,13 @@ public LocalDiskShuffleDataIO(SparkConf sparkConf) {
   }
 
   @Override
-  public ShuffleExecutorComponents executor() {
-    return new LocalDiskShuffleExecutorComponents(sparkConf);
+  public ShuffleDriverComponents initializeShuffleDriverComponents() {
+    return new LocalDiskShuffleDriverComponents(sparkConf);
   }
 
   @Override
-  public ShuffleDriverComponents driver() {
-    return new LocalDiskShuffleDriverComponents();
+  public ShuffleExecutorComponents initializeShuffleExecutorComponents(
+      String appId, String execId, Map<String, String> extraConfigs) {
+    return new LocalDiskShuffleExecutorComponents(sparkConf);
```

Comment:
nit: ignoring `extraConfigs` - assert it is empty ?

Reply:
Don't think it's particularly important that it's empty and ignored - maybe just add a warning log.